### PR TITLE
board: stm32_min_dev: Add I2C support

### PR DESCRIPTION
**Pull request by explora26**
This commit adds I2C support to stm32_min_dev, verified with BME280

Signed-off-by: Harry Jiang <explora26@gmail.com>

### CODE REVIEW

File: boards/arm/stm32_min_dev/stm32_min_dev.yaml [hunk 1]
```diff
@@ -6,3 +6,6 @@ toolchain:
   - zephyr
   - gnuarmemb
 ram: 20
+supported:
+  - gpio
```
Comment (MaureenHelm) [inline boards/arm/stm32_min_dev/stm32_min_dev.yaml]: If you want to add gpio, do it in a separate commit.

File: boards/arm/stm32_min_dev/stm32_min_dev_defconfig [hunk 2]
```diff
@@ -35,3 +35,6 @@ CONFIG_CLOCK_STM32_AHB_PRESCALER=1
 # APB1 clock must not exceed 36MHz limit
 CONFIG_CLOCK_STM32_APB1_PRESCALER=2
 CONFIG_CLOCK_STM32_APB2_PRESCALER=1
+
+#enable I2C
+CONFIG_I2C=y
```
Comment (MaureenHelm) [inline boards/arm/stm32_min_dev/stm32_min_dev_defconfig]: @erwango, do you want this enabled by default?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please update your commit message to have a body.

File: boards/arm/stm32_min_dev/stm32_min_dev_defconfig (hunk 2 above)
Comment (ydamigos) [reply boards/arm/stm32_min_dev/stm32_min_dev_defconfig]: It should be removed. It depends on the application if the I2C should be enabled.

**Review by ydamigos — DISMISSED** [review]

File: boards/arm/stm32_min_dev/stm32_min_dev.yaml (hunk 1 above)
Comment (explora26) [reply boards/arm/stm32_min_dev/stm32_min_dev.yaml]: OK, I will send another PR

File: boards/arm/stm32_min_dev/stm32_min_dev_defconfig (hunk 2 above)
Comment (explora26) [reply boards/arm/stm32_min_dev/stm32_min_dev_defconfig]: Agree with ydamigos, I removed this.

**Review by erwango — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]
